audioipc/src/ipccore.rs

```diff
@@ -630,21 +631,25 @@ where
 
         // Repeatedly grab outgoing items from the handler, passing each to `encode` for serialization into `outbound`.
         while let Some(mut item) = self.handler.produce()? {
-            item.prepare_send_message_handle(|handle, _target| {
-                // On Unix, the handle is encoded into a cmsg buffer for out-of-band transport via sendmsg.
-                #[cfg(unix)]
-                {
-                    cmsg::encode_handle(&mut outbound.cmsg, handle);
-                    Ok(handle)
-                }
-                // On Windows, the handle is transferred by duplicating it into the target remote process during message send.
+            let handle = if item.has_associated_handle() {
```

Comment:
At the IPC server level, the handle remoting can work in either direction.  It's true that the cubeb remoting layer only uses this to send handles in the server->client direction, but that's a cubeb-remoting level decision.  The current code layout is confusing since we're using message.rs for core IPC and the cubeb layer - refactoring those out would help a lot.